GTFS_converter.py

```diff
@@ -0,0 +1,241 @@
+    else:
+        seen_sequences[station_sequence] = len(identical_trip_groups)
+        identical_trip_groups.append([ordered_list_of_trips[i]])
+
+#this part gets the average travel times between station pairs
+df_stoptimes.to_csv('modified_stop_times.txt', sep=',', index=False) #temp file
+with open("modified_stop_times.txt", "r") as file:
+    reader = csv.DictReader(file)
+    travel_times = defaultdict(list)
+    prev_stop = {}
+    for row in reader:
+        trip_id = row['trip_id']
+        stop_id = row['stop_id']
+        sequence = int(row['stop_sequence'])
+        arrival_time = row['arrival_time']
+    
+        if prev_stop!={}:
+            if sequence > 1 and prev_stop['trip_id'] == trip_id:
+                pair = (prev_stop['stop_id'], stop_id)
+                hours, minutes, seconds = map(int, arrival_time.split(':'))    
+                arrival_time_in_seconds = hours * 3600 + minutes * 60 + seconds
+                hours, minutes, seconds = map(int, prev_stop['departure_time'].split(':'))    
+                departure_time_in_seconds = hours * 3600 + minutes * 60 + seconds
+                travel_time=arrival_time_in_seconds-departure_time_in_seconds
+                travel_times[pair].append(travel_time)
+    
+        prev_stop = {
+            'trip_id': trip_id,
+            'stop_id': stop_id,
+            'departure_time': row['departure_time'],
+            'sequence': sequence
+        }
+    average_travel_times = [[pair[0], pair[1], sum(times)/len(times)] for pair, times in travel_times.items()]
+    file.close()
+    os.remove('modified_stop_times.txt')
+
+#here the dictionaries of stations and edges are created 
+routes_list=[]
+processed_list_of_stations=[]
+compiled_list_of_routes=[]
+for list_of_stations in ordered_list_of_stations: #here we create the database of stations
+    if list_of_stations in processed_list_of_stations: #if the path has already been processed
+        routes_list=[]
+        continue
+    elif list_of_stations[::-1] in processed_list_of_stations: #if the reverse of this path has been processed, a second track is added 
+        for index, station in enumerate(list_of_stations):
+            stations['Station'+str(counter)]={'type':'SimpleStation','name':str(station)+"_2",'u':str(station)+"_1",'v':str(station)+"_2",'key':2,'args':{'mean_boarding': 20,'mean_alighting': 20}}
+            routes_list.append(['u:'+str(station)+"_2",'v:'+str(station)+"_1",'key:'+"2"])
+            if index==0:
+                pass
+            else:
+                previous_station=list_of_stations[index-1]
+                for travel_time in average_travel_times:
+                    if travel_time[0]==previous_station and travel_time[1]==station:
+                        average_travel_time=travel_time[2]
+                        break
+                edges['Edge'+str(counter)]={'type':'TimedTrack','u':str(previous_station)+"_2",'v':str(station)+"_1",'key':2,"traversal_time": average_travel_time}

```

Comment:
Traversal time should be an integer. Recommend `int(round(average_travel_time))`

Reply:
Oh also traversal times need to be nested under "args":

Should look like this:

```
    {
        "type": "TimedTrack",
        "u": "NB Shawnessy CTrain Station_2",
        "v": "NB Fish Creek - Lacombe CTrain Station_1",
        "key": 1,
        "args": {
            "traversal_time": 120.0
        }
    },
```